enzyme/Enzyme/MLIR/Implementations/LLVMAutoDiffOpInterfaceImpl.cpp

```diff
@@ -77,10 +77,18 @@ class PointerTypeInterface
     return builder.create<LLVM::NullOp>(loc, self);
   }
 
+  Value createAddOp(Type self, OpBuilder &builder, Location loc, Value a, Value b) const {
+    llvm_unreachable("TODO");
+  }
+
   Type getShadowType(Type self, unsigned width) const {
     assert(width == 1 && "unsupported width != 1");
     return self;
   }
+
+  bool isPointerType(Type self) const{
+    return false;
+  }
```

Comment:
This is the interface implementation for `LLVMPointerType`, are you sure it's not a pointer? :)